codersdk/deployment.go

```diff
@@ -1270,26 +1268,6 @@ when required by your organization's security policy.`,
 			Group:       &deploymentGroupProvisioning,
 			YAML:        "daemonsEcho",
 		},
-		{
-			Name:        "Poll Interval",
-			Description: "Time to wait before polling for a new job.",
-			Flag:        "provisioner-daemon-poll-interval",
-			Env:         "CODER_PROVISIONER_DAEMON_POLL_INTERVAL",
-			Default:     time.Second.String(),
-			Value:       &c.Provisioner.DaemonPollInterval,
-			Group:       &deploymentGroupProvisioning,
-			YAML:        "daemonPollInterval",
-		},
-		{
-			Name:        "Poll Jitter",
-			Description: "Random jitter added to the poll interval.",
-			Flag:        "provisioner-daemon-poll-jitter",
-			Env:         "CODER_PROVISIONER_DAEMON_POLL_JITTER",
-			Default:     (100 * time.Millisecond).String(),
-			Value:       &c.Provisioner.DaemonPollJitter,
-			Group:       &deploymentGroupProvisioning,
-			YAML:        "daemonPollJitter",
-		},
```

Comment:
If we remove these from the deployment config then it becomes an error to specify them at all. Could we keep them around for a couple releases but mark them as deprecated?

Reply:
We have deprecation for options that are replaced by something else.  In this case there is no replacement -- these options are just not needed any more.

I don't feel like waiting a few releases to remove these options will do us much good.  We don't have a good mechanism to get `coder server` CLI warnings in front of operator eyeballs.  It just drops some logs at start of day and they'll get buried.  We're just as likely to break people when we remove them later.

I could just change the Descriptions to "This flag is deprecated and unused" and leave them in.  I don't think there will ever be a better time than now to delete them.

WDYT?

Reply:
Thinking on this some more, this will probably only break folks who are specifying the arguments via CLI flags. I think if they're specified via environment variables they'll just get ignored. And we don't specify CLI args in our Helm chart, so the blast radius doesn't appear to be as large. And folks who are running Coder via the installation script would probably be specifying configuration knobs as environment variables in `/etc/coder.d/coder.env`.

So maybe it's OK to just remove them? 🤷

Thinking outside the scope of the PR, we could have a "Deprecated options" section in the deployment healthcheck JSON that lights up red if you have deprecated options specified.